File: Project Euler/p5.cpp
#include <bits/stdc++.h>

using namespace std;

long gcd(long x, long y) {
    long r = max(x, y) % min(x, y);
    if (r == 0) {  
        return min(x, y);
    } else {
        return gcd(min(x, y), r);
    }
}

int main() {
    long multiple = 1;
    for (long i = 1; i < 21; i++) {
        long divisor = gcd(multiple, i);
        if (divisor == 1) {
            multiple *= i;
        } else {
            multiple *= (i / divisor);
        }
    }
    cout << multiple << endl;
    return 0;
}

File: Project Euler/p5.py
# Smallest multiple
def gcd(x, y):
    r = max(x, y) % min(x, y)
    if r == 0:
        return min(x, y)
    return gcd(min(x, y), r)

def main():
    multiple = 1 
    for i in range(1, 21):
        div = gcd(multiple, i)
        if div == 1:
            multiple *= i
        else:
            multiple *= (i // div)
    return multiple

if __name__ == "__main__":
    print(main())
